src/sknnr/_base.py

```diff
@@ -13,6 +13,22 @@
     from .transformers._base import ComponentReducerMixin
 
 
+def _validate_data(estimator, *args, **kwargs):
+    """
+    Compatibility wrapper around sklearn's _validate_data function.
+
+    scikit-learn >= 1.6.0 moved _validate_data from a method of BaseEstimator to a
+    public utility function. This function wraps the utility function if available,
+    and falls back to the method if not.
+    """
+    try:
+        from sklearn.utils.validation import validate_data
+    except ImportError:
+        return estimator._validate_data(*args, **kwargs)
+
+    return validate_data(estimator, *args, **kwargs)
+
+
```

Comment:
Makes total sense.  Is there a standard way to leave "breadcrumbs" for ourselves when we move to `scikit-learn>=1.6.0`?  I would think we wouldn't need this function anymore at that point.

Reply:
I'm not aware of anything more authoritative than just leaving a "TODO" comment, which I'll add.